File: 2346/2346.cpp
/* ************************************************************************** */
/*                                                                            */
/*                                                        :::      ::::::::   */
/*   2346.cpp                                           :+:      :+:    :+:   */
/*                                                    +:+ +:+         +:+     */
/*   By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+        */
/*                                                +#+#+#+#+#+   +#+           */
/*   Created: 2024/03/03 11:20:55 by jihoolee          #+#    #+#             */
/*   Updated: 2024/03/03 13:21:41 by jihoolee         ###   ########.fr       */
/*                                                                            */
/* ************************************************************************** */

#include <iostream>
#include <deque>
#include <utility>

void get_input(std::deque<std::pair<int, int> >* d) {
  int cnt;
  int num;

  std::cin >> cnt;
  for (int i = 0; i < cnt; ++i) {
    std::cin >> num;
    d->push_back(std::make_pair(i + 1, num));
  }
}

void rotate_deque(std::deque<std::pair<int, int> >* d, int rotation) {
  if (rotation > 0) {
    for (int i = 0; i < rotation - 1; ++i) {
      d->push_back(d->front());
      d->pop_front();
    }
  } else {
    for (int i = 0; i > rotation; --i) {
      d->push_front(d->back());
      d->pop_back();
    }
  }
}

int main(void) {
  std::deque<std::pair<int, int> >  d;
  int                               rotation;

  get_input(&d);
  do {
    std::cout << d.front().first << ' ';
    rotation = d.front().second;
    d.pop_front();
    rotate_deque(&d, rotation);
  } while (!d.empty());
  return 0;
}

File: 2346/2346.py
# **************************************************************************** #
#                                                                              #
#                                                         :::      ::::::::    #
#    2346.py                                            :+:      :+:    :+:    #
#                                                     +:+ +:+         +:+      #
#    By: jihoolee <jihoolee@student.42SEOUL.kr>     +#+  +:+       +#+         #
#                                                 +#+#+#+#+#+   +#+            #
#    Created: 2024/03/03 12:46:05 by jihoolee          #+#    #+#              #
#    Updated: 2024/03/03 13:24:04 by jihoolee         ###   ########.fr        #
#                                                                              #
# **************************************************************************** #

from collections import deque


def rotate_deque(d: deque, rotation: int):
    if rotation > 0:
        for _ in range(rotation - 1):
            d.append(d.popleft())
    else:
        for _ in range(-rotation):
            d.appendleft(d.pop())


def main() -> None:
    _ = input()
    nums: deque = deque(list(enumerate(map(int, input().split()))))
    result: list[int] = []

    while True:
        result.append(nums[0][0] + 1)
        rotation: int = nums[0][1]
        nums.popleft()
        if len(nums) == 0:
            break
        rotate_deque(nums, rotation)
    print(*result)


if __name__ == '__main__':
    main()
